### ISSUE TITLE
Using GTK+3 with Quartz on OS X using homebrew.

### ISSUE THREAD

**Issue by oderwat** [StefanSalewski/nim-gtk3#1]
Hi. Just to inform you about some success on OS X!

I just tried to use your Nim GTK+3 bindings with -d:gtk_quartz on my mac.

I installed GTK+3 and its dependencies using homebrew simply like this:

```
brew install gtk+3 --without-x11
```

Then I tried to compile the test.nim in gtk-3 test directory. This failed because `libgtk-quartz-3.0.dylib` was not found!

It seems that this lib for GTK 3 is just called `libgtk-3.0.dylib` now!

After I changed the name in the gtk3.nim (line 6) I could compile and run the demo and running it I got a nice little "native" window!

Maybe you want to include this info in the readme and change the name of the lib. But I am going to make a PR later today too.

P.S.: To clone (or update) and install all the wrappers I created this little (simple) script. Maybe somebody has use for that: https://gist.github.com/oderwat/b56c0f0e5c7dbc54ae67

**Comment by StefanSalewski** [comment]
Thanks for the hint. I am a bit surprised that there is interest for GTK3 on Mac at all. I will fix the library name. Unfortunately there are many libraries involved in my GTK3 related wrappers, I have 10 modules now, and lib names for Windows and Mac are only guesses. For gtk2 modules there was something like 

 elif defined(gtk_quartz):
  const LIB_GTK* = "libgtk-quartz-3.0.dylib"
elif defined(macosx):
  const LIB_GTK* = "libgtk-x11-3.0.dylib"

in the module, I copied that. I am not sure what the difference between macosx  and quartz is, and I am not sure also if such distinction makes sense for the other related libraries. Currently I am going to upload GIO and ATK modules...

**Comment by oderwat** [comment]
My interest in GTK3 is because of cross development for Windows from Mac and Ubuntu-Linux in this case. I would prefer wxWidgets though but don't have the resources to make a wrapper for this myself. I checked out some GTK3 apps which are available on Mac / Win an Linux and even when they are ugly as hell it is a solution for the problems we need to solve.

**Comment by oderwat** [comment]
The "quartz" GKT uses more native windows and menus without starting the X11 windows manager. The "real" x11 is available as third party software on Mac but does not fit the native window management on Mac very well.

**Comment by StefanSalewski** [comment]
Yes, I have to admit that GTK3 has some problems on Windows and Mac. For Linux we have soon 3.16, and it is really not bad for Linux. For Windows there seems to be only 3.6.4 as a binary available. And, I recently discovered that even latest Inkscape 0.91 uses still GTK2 for Windows. I think the reason for bad GTK3 support is, that GTK development is done by Gnome developers now, and they do not really care about Windows and Mac. Personally I can understand that well, for me it makes not much sense to use a propritary systems like Windows or Mac with open source software. Currently I do not use Windows or mac at all, but if I would, I would not use it together with open source software. If you really want to test GTK3 for windows -- I was told that there exists a GTK download manager which uses GTK3 on windows, see  http://ugetdm.com/ . For commercial Software GTK generally is not a good choice, people do not like it for Windows and Mac.

**Comment by oderwat** [comment]
You sound as if open-source software is limited to linux. There is a lot of that for mac and windows too.
GTK is just perceived as very ugly looking/behaving, It was not developed for using it on mac and windows. I also don't think that it is primarily the Windows or Mac users which want GTK to run on their system. It seems more like the developers want to spread their software to those systems. That said I would use those software on linux. I usually run a ubuntu and windows machines on my macs anyway.
Well. I am not yet sure what we need to use. But GTK3 is an option.

**Comment by oderwat** [comment]
For testing it on windows I think more about evaluating this use case: http://www.gnucash.org/

**Comment by StefanSalewski** [comment]
Is GNUCASH already using GTK3 on Windows? If so, please let me know. I heard that the next version of Inkscape should also use GTK3 on Windows, but it may take some time. Inkscape seems to be the most advanced GTK application -- but still some people do not like it visual appearance.

For the GUI toolkit: I just discussed the state of Qt bindings for other languages with some people. Seems that state is not really good for all the compiled languages, i.e. Haskell bindings are death as someone told me, D seems to be not really active also. And I think making Qt bindings for Nim or Rust would be a very large effort. And wxWidgets -- I would never really consider using that. I was told API is bad, only very few applications are using it. A GUI coded in Nim would be nice of course, but I think we have not the manpower for that. Yesterday I discovered that there exists a native GUI toolkit for D-lang called DlangUI, http://wiki.dlang.org/GUI_Libraries, maybe a starting point?

**Comment by oderwat** [comment]
We use wxWidgets in multiple applications we sell to customers and it "just works". The same application works on Windows 8.1 and Windows XP / 98 / NT and at the same time and looks good. It didn't change for nearly 10 years now.

It also works on Linux and Mac if needed. But here we just cross developed most of that on Windows first, later on OS X and run it in the Windows XP VM. The code is python though. I wanted to replace this with Nim for different reasons.

To me it is the superior GUI toolkit because: "Whenever possible, wxWidgets uses the native platform SDK and system provided widgets. This means that a program compiled on Windows will have the look and feel of a Windows program, and when compiled on a Linux machine, it will get the look and feel of a Linux program. "

I do not really care about what others use (esp. in Open-Source projects). The Interface worked for what we needed it in the last 10 years and we even have some custom Widgets in use. I also contributed a bit to their tooling (long ago).

I did not check if GNU-Cash uses GTK3 though (it uses 2). I just checked if GTK3 it is available for Windows and estimated that it will not look much worse than that :). We could simply use GTK2 if we need to and no other options are available. In the end we could also live with an "Browser"-Frontend (using Riot possibly) for some of the projects.

I think a native Nim GUI with cross systems native widgets will never happen. I guess there will be some game (including mobile usage) related GUI implementations eventually.
